api/metadata_handler.go

```diff
@@ -51,16 +49,106 @@ func apiMetadataHandler(w http.ResponseWriter, r *http.Request) {
 		shared.CacheStatusOK).ServeHTTP(w, r)
 }
 
+func apiMetadataTriageHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := shared.NewAppEngineContext(r)
+	ds := shared.NewAppEngineDatastore(ctx, false)
+	user, token := shared.GetUserFromCookie(ctx, ds, r)
+	if user == nil || token == nil {
+		http.Error(w, "User is not logged in", http.StatusBadRequest)
+		return
+	}
+
+	githubBotClient, err := getWPTFYIGithubBot(ctx)
+	if err != nil {
+		http.Error(w, "Unable to get Github Client: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	aeAPI := shared.NewAppEngineAPI(ctx)
+	git := shared.GetMetadataGithub(githubBotClient, user.GitHubHandle, user.GithuhEmail)
+	tm := shared.GetTriageMetadata(ctx, git, shared.GetLogger(ctx), aeAPI.GetHTTPClient())
+
+	gac := shared.NewGitAccessControl(ctx, ds, *token)
+	handleMetadataTriage(ctx, gac, tm, w, r)
+}
+
+func handleMetadataTriage(ctx context.Context, gac shared.GitHubAccessControl, tm shared.TriageMetadataInterface, w http.ResponseWriter, r *http.Request) {
+	if r.Method != "PATCH" {
+		http.Error(w, "Invalid HTTP method; only accpet PATCH request", http.StatusBadRequest)
+		return
+	}
+
+	contentType := r.Header.Get("Content-Type")
+	if contentType != "application/json" {
+		http.Error(w, "Invalid content-type: %s"+contentType, http.StatusBadRequest)
+		return
+	}
+
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "Failed to read PATCH request body", http.StatusInternalServerError)
+		return
+	}
+
+	err = r.Body.Close()
+	if err != nil {
+		http.Error(w, "Failed to finish reading request body", http.StatusInternalServerError)
+		return
+	}
+
+	var metadata shared.MetadataResults
+	err = json.Unmarshal(data, &metadata)
+	if err != nil {
+		http.Error(w, "Failed to parse JSON: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Check if the token is still valid.
+	code, err := gac.IsValidAccessToken()
+	if err != nil {
+		http.Error(w, "Fail to validate user token "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if code != http.StatusOK {
+		http.Error(w, "User token invalid; please log in again. ", http.StatusBadRequest)
+		return
+	}
+
+	code, err = gac.IsValidWPTMember()
+	if err != nil {
+		http.Error(w, "Fail to validate web-platform-tests membership: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if code != http.StatusOK {
+		http.Error(w, "User is not a part of web-platform-tests org", http.StatusBadRequest)
```

Comment:
It may not be obvious to someone who gets this error what they have to do. I would suggest something like:

> Logged-in user must be a member of the web-platform-tests GitHub organization. To join, visit XYZ and do Foo, Bar.

(Or whatever the most succinct description is)